neuralforecast/tsdataset.py

```diff
@@ -370,12 +408,178 @@ def __init__(
+
+    def __init__(
+        self,
+        files_ds: List[str],
+        temporal_cols,
+        id_col: str,
+        time_col: str,
+        target_col: str,
+        last_times,
+        indices,
+        max_size: int,
+        min_size: int,
+        y_idx: int,
+        static=None,
+        static_cols=None,
+        sorted=False,
+    ):
+        super().__init__(
+            temporal_cols=temporal_cols,
+            max_size=max_size,
+            min_size=min_size,
+            y_idx=y_idx,
+            static=static,
+            static_cols=static_cols,
+            sorted=sorted,
+        )
+        self.files_ds = files_ds
+        self.id_col = id_col
+        self.time_col = time_col
+        self.target_col = target_col
+        # array with the last time for each timeseries
+        self.last_times = last_times
+        self.indices = indices
+        self.n_groups = len(files_ds)
+
+    def __getitem__(self, idx):
+        if not isinstance(idx, int):
+            raise ValueError(f"idx must be int, got {type(idx)}")
+
+        temporal_cols = self.temporal_cols
+        data = pd.read_parquet(self.files_ds[idx], columns=temporal_cols).to_numpy()
+        data, temporal_cols = TimeSeriesDataset._ensure_available_mask(
+            data, temporal_cols
+        )
+        data = self._as_torch_copy(data)
+
+        # Pad the temporal data to the left
+        temporal = torch.zeros(
+            size=(len(temporal_cols), self.max_size), dtype=torch.float32
+        )
+        temporal[: len(temporal_cols), -len(data) :] = data.permute(1, 0)
+
+        # Add static data if available
+        static = None if self.static is None else self.static[idx, :]
+
+        item = dict(
+            temporal=temporal,
+            temporal_cols=temporal_cols,
+            static=static,
+            static_cols=self.static_cols,
+            y_idx=self.y_idx,
+        )
+
+        return item
+
+    @staticmethod
+    def from_data_directories(
+        directories,
+        static_df=None,
+        sort_df=False,
+        exogs=[],
+        id_col="unique_id",
+        time_col="ds",
+        target_col="y",
+    ):
+        """We expect directories to be a list of directories of the form [unique_id=id_0, unique_id=id_1, ...]. Each directory should contain the timeseries corresponding to that unqiue_id,
+        represented as a pandas or polars DataFrame. The timeseries can be entirely contained in one parquet file or split between multiple, but within each parquet files the timeseries should be sorted by time.
+        Static df should also be a pandas or polars DataFrame"""
+        import pyarrow as pa
+
+        # Define indices if not given and then extract static features
+        static, static_cols = TimeSeriesDataset._extract_static_features(
+            static_df, sort_df, id_col
+        )
+
+        max_size = 0
+        min_size = float("inf")
+        last_times = []
+        ids = []
+        exogs = set(exogs)
```

Comment:
You can use a different variable here just to check that the files have the expected temporal columns.
```suggestion
        expected_temporal = {target_col, *exogs}
```